client/src/pages/DirectoryPage/DirectoryPeople.js

```diff
@@ -16,173 +16,324 @@ import image4 from "../../assets/images/profile-picture-4.png";
+      arr.push(
+        <div className="CommunityPage__momCard-single">
+        <PersonCard image={this.state.others[i].image} name={this.state.others[i].first_name} location="Washington D.C." />
+        <button className="FriendButton" onClick={this.sendRequest}>Add Friend</button>
+        </div>
+      )
+    }
+  }
+
+  showRequests = () => {
+    let cards = []
+    this.requestCards(cards)
+    return cards
+  }
+
+  requestCards = (arr) => {
+    let length  = friendRequests.length
+    for (let i = 0; i < length; i++) {
+      //same as the friends function
+      arr.push(
+        <div className="CommunityPage__momCard-single">
+        <PersonCard image={this.state.friendRequests[i].image} name={this.state.friendRequests[i].first_name} location="Washington D.C." />
+        <button className="FriendButton" onClick={this.acceptRequest}>Accept</button>
+        </div>
+      )
+    }
+  }
+
+  trueFriends = (friendId) => {
+    for (let i = 0; i < this.props.connections.length; i++) {
+      if (this.props.connections[i] === friendId) {
+        return true
+      }
+    }
+  }
+
+  removeFriend = (e) => {
+    let nonFriend = e.target.parentElement.children[0].children[1].textContent
+    let connectionsList = this.props.connections
+    for (let i = 0; i < this.state.friends.length; i++) {
+      if (this.state.friends[i].first_name === nonFriend) {
+        connectionsList.splice(i, 1)
+      }
+    }
+    let data = { connections: connectionsList }
+    fetch(`http://127.0.0.1:8000/profile/${this.props.userid}`, {
+      method: 'PATCH',
+      headers: {
+        Authorization: `Token ${localStorage.getItem("token")}`,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data)
+    })
+    .then(response => response.json())
+    .then(result => {
+      console.log('Success:', result);
+    })
+    .catch(error => {
+      console.error('Error:', error);
+    });
+    window.location.reload(true)
+  }
+
+  sendRequest = (e) => {
+    let newFriend = e.target.parentElement.children[0].children[1].textContent
+    e.target.parentElement.children[1].textContent = "Request Sent"
+    let newFriendId
+    let requestList
+    for (let i = 0; i < this.state.others.length; i++) {
+      if (this.state.others[i].first_name === newFriend) {
+        newFriendId = this.state.others[i].user
+        requestList = this.state.others[i].requests
+      }
+    }
+    requestList.push(this.props.userid)
+    let data = { requests: requestList }
+    fetch(`http://127.0.0.1:8000/profile/${newFriendId}`, {
+      method: 'PATCH',
+      headers: {
+        Authorization: `Token ${localStorage.getItem("token")}`,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data)
+    })
+    .then(response => response.json())
+    .then(result => {
+      console.log('Success:', result);
+    })
+    .catch(error => {
+      console.error('Error:', error);
+    });
+  }
```

Comment:
You should get the users's id that the currently signed in user wants to friend and send a request to the backend with just that. All you need is the user ID for the friend request and the currently signed in user